# Formula Student Driverless Project (FSD-Project).
# Copyright (c) 2022:
#  - FengYunJi <yunjifeng@bitfsd.cn>
# FSD-Project is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# FSD-Project is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# You should have received a copy of the GNU General Public License
# along with FSD-Project.  If not, see <https://www.gnu.org/licenses/>.

Subscriber:

    camera_topic_name: /camera/image_raw

Publisher:

    detected_image_topic_name: /ppyolo/detection_image
    
    boundingbox_topic_name: /ppyolo/bounding_boxes
    
    object_topic_name: /ppyolo/found_object

Parameter:

    type: paddledetection # [paddlex, paddledetection]

    use_gpu: True

    use_tensorrt: True

    tensorrt_type: FP16 # [FP16, FP32, INT8]

    model_dir: /home/feng1909/ConeDetection/src/ppyolo/scripts/ppyolov2_r50vd_dcn_voc/
    # model_dir = '/home/feng1909/ConeDetection/src/ppyolo/scripts/inference_model/'
    # model_dir = '/home/feng1909/ConeDetection/src/ppyolo/scripts/yolov3_r50vd_dcn_270e_coco/'

    img_size: 480

    node_rate: 30


